File: Objective-C/Reference App/205. Search UI/DisplaySearchViewController.m
//
//  DisplaySearchViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "DisplaySearchViewController.h"

@interface DisplaySearchViewController ()

// Put this inside your @interface
@property (nonatomic, nullable) UINavigationController *searchNavController;

@end

@implementation DisplaySearchViewController

- (IBAction)go:(id)sender {
    [self showSearch];
}

// Put this inside your @implementation
- (void)showSearch {
    TGSearchViewController *searchVC = [TGSearchViewController create];
    searchVC.delegate = self;
    
    UINavigationController *navController = [searchVC embedInNavigationController];
    [self presentViewController:navController animated:YES completion:nil];
    
    self.searchNavController = navController;
}

#pragma mark - TGSearchViewControllerDelegate

- (CLLocationCoordinate2D)exampleCoordinate {
    return CLLocationCoordinate2DMake(34.101558, -118.340944); // Grauman's Chinese Theatre
}

- (TGSearchExtent *)exampleSearchExtent {
    return [TGSearchExtent.alloc initWithSouthwestCoordinate:CLLocationCoordinate2DMake(34.055534, -118.37313)
                                         northeastCoordinate:CLLocationCoordinate2DMake(34.1504, -118.308757)];
}

- (CLLocationCoordinate2D)coordinateForRequestType:(TGSearchViewControllerRequestType)requestType {
    return self.exampleCoordinate;
    
    // Note: If you have a TGMapView or MGLMapView, you can get the current location like this:
    //return self.mapView.centerCoordinate;
}

- (TGSearchExtent *)searchExtentForRequestType:(TGSearchViewControllerRequestType)requestType {
    return self.exampleSearchExtent;
    
    // Note: If you have a TGMapView or MGLMapView, you can get the current bounds like this:
    //return [TGSearchExtent.alloc initWithCoordinateBounds:self.mapView.visibleCoordinateBounds];
}

- (void)didSelectSearchResult:(TGSearchResult *)searchResult {
    [self.searchNavController dismissViewControllerAnimated:YES completion:nil];
    
    // Do something with the selected search result
    
    [self handleOutput:searchResult];
}

@end

File: Swift 4/Reference App/5. Display Search/DisplaySearchViewController.swift
//
//  DisplaySearchViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class DisplaySearchViewController: OutputViewController, TGSearchViewControllerDelegate {

    @IBAction func go(_ sender: Any) {
        showSearch()
    }
    
    // MARK: -
    
    var searchNavController: UINavigationController?
    
    func showSearch() {
        let searchVC = TGSearchViewController.create()
        searchVC.delegate = self
        
        let navController = searchVC.embedInNavigationController()
        present(navController, animated: true)
        
        searchNavController = navController
    }
    
    let exampleCoordinate = CLLocationCoordinate2D(latitude: 34.101558, longitude: -118.340944) // Grauman's Chinese Theatre
    let exampleSearchExtent = TGSearchExtent(southwestCoordinate: CLLocationCoordinate2D(latitude: 34.055534, longitude: -118.37313),
                                             northeastCoordinate: CLLocationCoordinate2D(latitude: 34.1504, longitude: -118.308757))
    
    func coordinate(for requestType: TGSearchViewControllerRequestType) -> CLLocationCoordinate2D {
        return exampleCoordinate
        
        // Note: If you have a TGMapView or MGLMapView, you can get the current location like this:
        //return mapView.centerCoordinate
    }
    
    func searchExtent(for requestType: TGSearchViewControllerRequestType) -> TGSearchExtent {
        return exampleSearchExtent
        
        // Note: If you have a TGMapView or MGLMapView, you can get the current bounds like this:
        //return TGSearchExtent(coordinateBounds: mapView.visibleCoordinateBounds)
    }
    
    func didSelect(_ searchResult: TGSearchResult) {
        searchNavController?.dismiss(animated: true)
        
        // Do something with the selected search result
        
        handleOutput(searchResult)
    }

}
